src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSource.java

```diff
@@ -266,18 +266,19 @@ protected SCMRevision retrieve(@NonNull SCMHead head, @NonNull TaskListener list
                 MergeRequest mr =
                     gitLabApi.getMergeRequestApi()
                         .getMergeRequest(gitlabProject, Integer.parseInt(h.getId()));
+                String targetSha = gitLabApi.getRepositoryApi().getBranch(mr.getTargetProjectId(), mr.getTargetBranch()).getCommit().getId();
                 if (mr.getState().equals(Constants.MergeRequestState.OPENED.toString())) {
                     listener.getLogger().format("Current revision of merge request #%s is %s%n",
                         h.getId(), mr.getDiffRefs().getHeadSha());
```

Comment:
shouldn't this occurence of `mr.getDiffRefs().getHeadSha()` be also changed to `mr.getSha()`?